src/aws-secrets-controller/src/lib.rs

```diff
@@ -209,15 +211,22 @@ impl AwsSecretsClient {
 #[async_trait]
 impl SecretsReader for AwsSecretsClient {
     async fn read(&self, id: GlobalId) -> Result<Vec<u8>, anyhow::Error> {
-        Ok(self
-            .client
-            .get_secret_value()
-            .secret_id(self.secret_name(id))
-            .send()
-            .await?
-            .secret_binary()
-            .ok_or_else(|| anyhow!("internal error: secret missing secret_binary field"))?
-            .to_owned()
-            .into_inner())
+        info!(%id, "reading secret from AWS");
```

Comment:
Do you want to stick a unique id (e.g. a UUID) in in case two happen concurrently?

Reply:
Ooh, good thinking.

Reply:
> in in case

case in point 😄

Reply:
Good idea. Done!